app/controllers/merchant/items_controller.rb

```diff
@@ -30,19 +31,19 @@ def toggle
 
   def create
     @merchant = Merchant.find(current_user.merchant_id)
-    item = @merchant.items.create(item_params)
-    if item.save
+    @item = @merchant.items.create(item_params)
```

Comment:
We might be able to refactor this later so that we don't need to require [@merchant, @item] in the form for, because the item is already connected to the merchant